docs/vfio-user.rst

```diff
@@ -1127,6 +1140,111 @@ client must write data on the same order and transction size as read.
 If an error occurs then the server must fail the read or write operation. It is
 an implementation detail of the client how to handle errors.
 
+VFIO_USER_DEVICE_GET_REGION_IO_FDS
+----------------------------------
+
+Message format
+^^^^^^^^^^^^^^
+
++--------------+------------------------+
+| Name         | Value                  |
++==============+========================+
+| Message ID   | <ID>                   |
++--------------+------------------------+
+| Command      | FIXME                      |
++--------------+------------------------+
+| Message size | FIXME          |
++--------------+------------------------+
+| Flags        | Reply bit set in reply |
++--------------+------------------------+
+| Error        | 0/errno                |
++--------------+------------------------+
+| Region IO info  | Region IO fd info       |
++--------------+------------------------+
+
+This command message is sent by the client to the server to query for
+sub-regions of the given region that should be accessed via file descriptors.
+
+A client should hook up the returned file descriptors as requested; when
+accesses are made to these sub-regions, the file descriptor is directly
+notified/written (for example, by KVM), avoiding a round-trip through the client
+and a VFIO_USER_REGION_READ/WRITE call.
+
+
+Region IOFD info format
+^^^^^^^^^^^^^^^^^^^^^^^
+
++------------+--------+------------------------------+
+| Name       | Offset | Size                         |
++============+========+==============================+
+| argsz      | 16     | 4                            |
++------------+--------+------------------------------+
+| flags      | 20     | 4                            |
++------------+--------+------------------------------+
+|            | +-----+-----------------------------+ |
+|            | | Bit | Definition                  | |
+|            | +=====+=============================+ |
+|            | | 0   | FIXME
+|            | +-----+-----------------------------+ |
++------------+--------+------------------------------+
+| index      | 24     | 4                            |
++------------+--------+------------------------------+
+| nr_subregions     | 28     | number of subregions in array |
++------------+--------+------------------------------+
+| subregions | 32     | array of subregions          |
++------------+--------+------------------------------+
+
+* *index* is the index of memory region being queried, it is the only field
```

Comment:
nit: the order of each element in the list should match the order in the definition above

Reply:
done, thanks